php/libraries/NDB_Config.class.inc

```diff
@@ -396,12 +396,40 @@ class NDB_Config
             $XMLValue = $this->getSettingFromXML($name);
 
             if ($XMLValue !== null) {
+                switch($name) {
+                case 'database':
+                case 'sandbox':
+                case 'showDatabaseQueries':
+                    return $XMLValue;
+                }
+                $db = Database::singleton();
+                if ($db->isConnected() !== true) {
+                    // no database connection, fall back on config.xml
+                    throw new DatabaseException(
+                        "No connection to database while trying to get setting $name"
+                    );
+                }
+                $configSetting = $db->pselect(
+                    "SELECT child.ID,child.Name
+                    FROM ConfigSettings cs
+                        LEFT JOIN ConfigSettings child ON (child.Parent=cs.ID)
+                    WHERE cs.Name=:nm",
+                    array("nm" => $name)
+                );
+                foreach ($configSetting as $child) {
+                    if (!isset($XMLValue[$child['Name']])) {
+                        $XMLValue[$child['Name']] = $this->getSettingFromDB($child['Name'], $child['ID']);
```

Comment:
Travis is flagging this line - too long.  @jstirling91 should be super-quick fix